### PR TITLE
Average order value graph does not display any data

### LINKED ISSUE #7851: Average order value graph
Hi, i have a problem with graph for average order value in e-commerce sites. When all data is already available but we cannot see it on the graph. Take a look on sreenshot:

zrzut ekranu 2015-05-06 o 17 37 34

API result:

```
<result>
<nb_conversions>19</nb_conversions>
<nb_visits_converted>14</nb_visits_converted>
<revenue>13869.51</revenue>
<revenue_subtotal>5880.4</revenue_subtotal>
<revenue_tax>1361.07</revenue_tax>
<revenue_shipping>368.11</revenue_shipping>
<revenue_discount>686</revenue_discount>
<items>47</items>
<avg_order_revenue>729.97</avg_order_revenue> // <------
<conversion_rate>1.95%</conversion_rate>
<nb_conversions_new_visit>19</nb_conversions_new_visit>
<nb_visits_converted_new_visit>14</nb_visits_converted_new_visit>
<revenue_new_visit>13869.51</revenue_new_visit>
<revenue_subtotal_new_visit>5880.4</revenue_subtotal_new_visit>
<revenue_tax_new_visit>1361.07</revenue_tax_new_visit>
<revenue_shipping_new_visit>368.11</revenue_shipping_new_visit>
<revenue_discount_new_visit>686</revenue_discount_new_visit>
<items_new_visit>47</items_new_visit>
<avg_order_revenue_new_visit>729.97</avg_order_revenue_new_visit>
<conversion_rate_new_visit>3.41%</conversion_rate_new_visit>
<nb_conversions_returning_visit>0</nb_conversions_returning_visit>
<nb_visits_converted_returning_visit>0</nb_visits_converted_returning_visit>
<revenue_returning_visit>0</revenue_returning_visit>
<revenue_subtotal_returning_visit>0</revenue_subtotal_returning_visit>
<revenue_tax_returning_visit>0</revenue_tax_returning_visit>
<revenue_shipping_returning_visit>0</revenue_shipping_returning_visit>
<revenue_discount_returning_visit>0</revenue_discount_returning_visit>
<items_returning_visit>0</items_returning_visit>
<avg_order_revenue_returning_visit>0</avg_order_revenue_returning_visit>
<conversion_rate_returning_visit>0%</conversion_rate_returning_visit>
</result>
```

### PR DESCRIPTION
**Pull request by tsteur**
fixes  #7851

There were 2 problems:
- `avg_order_revenue` depends on `revenue` and `conversion_rate` but when selecting `avg_order_revenue` only that metric was fetched, the dependent ones were not fetched
- `avg_order_revenue` was formatted eg to `$ 12.04` which resulted in `0` when converting to float.

### CODE REVIEW

File: plugins/Goals/API.php [hunk 1]
```diff
@@ -402,6 +403,14 @@ public function getMetrics($idSite, $period, $date, $segment = false, $idGoal =
...
         $requestedColumns = Piwik::getArrayFromApiParameter($columns);
 
+        $shouldAddAverageOrderRevenue = (in_array('avg_order_revenue', $requestedColumns) || empty($requestedColumns)) && $isEcommerceGoal;
+
+        if ($shouldAddAverageOrderRevenue && !empty($requestedColumns)) {
+            $avgOrder = new AverageOrderRevenue();
+            $metricsToAdd = $avgOrder->getDependentMetrics();
+            $requestedColumns = array_unique(array_merge($requestedColumns, $metricsToAdd));
+        }
+
```
Comment (diosmosis) [inline plugins/Goals/API.php]: This logic should be handled by the Report::getMetricsRequiredForReport() call below, is there a reason it isn't?

Comment (diosmosis) [inline plugins/Goals/API.php]: Nevermind I see my comment below about this.

**Comment by diosmosis** [comment]
Should there be a new test for this? Existing tests don't change, so I assume this code path isn't being tested.

**Comment by tsteur** [comment]
Added a test

**Pull request merged**